internals/overlord/checkstate/manager.go

```diff
@@ -633,3 +644,50 @@ func (m *CheckManager) Replan() {
 		m.updateCheckData(check, changeID, 0)
 	}
 }
+
+// RefreshCheck runs a check immediately.
+func (m *CheckManager) RefreshCheck(ctx context.Context, check *plan.Check) (*CheckInfo, error) {
+	m.checksLock.Lock()
+	checkData := m.ensureCheck(check.Name)
+	refresh := checkData.refresh
+	result := checkData.result
+	m.checksLock.Unlock()
+
+	getCheckInfo := func() *CheckInfo {
+		m.checksLock.Lock()
+		checkData := m.ensureCheck(check.Name)
+		m.checksLock.Unlock()
```

Comment:
I believe we need to lock for this whole function, right? Because this is a pointer to `checkData`, it could be modified while we're looking at it (non-atomic). So I recommend the "lock...defer unlock" pattern.